### PR TITLE
Provide example config, improve pid check

### PR DESCRIPTION
**Pull request by theopolis**
Working towards better install/package experience on OSX. Also removing gotos for #579 and improving pid checking by including the basename in OSX processes table when a `proc_name` isn't available.

### CODE REVIEW

File: tools/deployment/osquery.example.conf [hunk 1]
```diff
@@ -0,0 +1,71 @@
+{
+  /* Configure the daemon below */
```
Comment (marpaia) [inline tools/deployment/osquery.example.conf]: Does the boost JSON parser work with comments in JSON?

File: osquery/core/system.cpp [hunk 2]
```diff
@@ -106,76 +106,67 @@ std::vector<fs::path> getHomeDirectories() {
...
+    boost::filesystem::remove(FLAGS_pidfile);
+  } catch (boost::filesystem::filesystem_error& e) {
+    // Unable to remove old pidfile.
+    LOG(WARNING) << "Unable to remove the osqueryd pidfile.";
+  }
+
+  return Status(0, "OK");
+}
+
 Status createPidFile() {
```
Comment (marpaia) [inline osquery/core/system.cpp]: This really needed work, thanks so much for doing this.

**Comment by theopolis** [comment]
Yeah, boost's JSON parser accepts C/C++ style comments. And you got it bud. :)

**Comment by marpaia** [comment]
That's awesome! I had no idea.

**Comment by marpaia** [comment]
we should write a simple osquery-config-check tool or something that just reads in the config and makes sure s.ok() == true. thoughts?

**Comment by marpaia** [comment]
previously, i would tell people to test their json at jsonlint.com, but with comments in our example, we can't do that anymore.

**Comment by theopolis** [comment]
Let's get an issue going that outlines all of the commands we'd want in a osqueryctl: enable/disable/config-check/status/info?

**Comment by marpaia** [comment]
on it

**Comment by marpaia** [comment]
we need @osqueryer to respond to #fileatask

**Comment by theopolis** [comment]
:cake:

**Comment by marpaia** [comment]
#585

**Pull request merged**